HeatStroke/Graphics/Particles/Emitter.cpp

```diff
@@ -0,0 +1,702 @@
+	InitAffectorVelocity();
+}
+
+void HeatStroke::Emitter::InitAffectorColor(tinyxml2::XMLElement* p_pAffectorColorElement)
+{
+#if _DEBUG
+	assert(p_pAffectorColorElement != nullptr);
+#endif
+
+	Affector::AFFECTOR_FUNCTION eAffectorFunction = Affector::AFFECTOR_FUNCTION::AF_CONSTANT;
+	std::string strAffectorFunction = "";
+	EasyXML::GetRequiredStringAttribute(p_pAffectorColorElement, "function", strAffectorFunction);
+	const char* cAffectorFunction = strAffectorFunction.c_str();
+	if (strcmp(cAffectorFunction, "lifetime") == 0)
+	{
+		eAffectorFunction = Affector::AFFECTOR_FUNCTION::AF_LIFETIME;
+	}
+
+	glm::vec3 vColor = glm::vec3(0.0f, 0.0f, 0.0f);
+	EasyXML::GetRequiredFloatAttribute(p_pAffectorColorElement, "r", vColor.r);
+	EasyXML::GetRequiredFloatAttribute(p_pAffectorColorElement, "g", vColor.g);
+	EasyXML::GetRequiredFloatAttribute(p_pAffectorColorElement, "b", vColor.b);
+	m_vAffectors.push_back(new AffectorColor(eAffectorFunction, vColor));
+}
+
+void HeatStroke::Emitter::InitAffectorFade(tinyxml2::XMLElement* p_pAffectorFadeElement)
+{
+#if _DEBUG
+	assert(p_pAffectorFadeElement != nullptr);
+#endif
+
+	Affector::AFFECTOR_FUNCTION eAffectorFunction = Affector::AFFECTOR_FUNCTION::AF_CONSTANT;
+	std::string strAffectorFunction = "";
+	EasyXML::GetRequiredStringAttribute(p_pAffectorFadeElement, "function", strAffectorFunction);
+	const char* cAffectorFunction = strAffectorFunction.c_str();
+	if (strcmp(cAffectorFunction, "lifetime") == 0)
+	{
+		eAffectorFunction = Affector::AFFECTOR_FUNCTION::AF_LIFETIME;
+	}
+
+	float fFade = 0.0f;
+	EasyXML::GetRequiredFloatAttribute(p_pAffectorFadeElement, "value", fFade);
+	m_vAffectors.push_back(new AffectorFade(eAffectorFunction, fFade));
+}
+
+void HeatStroke::Emitter::InitAffectorScale(tinyxml2::XMLElement* p_pAffectorScaleElement)
+{
+#if _DEBUG
+	assert(p_pAffectorScaleElement != nullptr);
+#endif
+
+	Affector::AFFECTOR_FUNCTION eAffectorFunction = Affector::AFFECTOR_FUNCTION::AF_CONSTANT;
+	std::string strAffectorFunction = "";
+	EasyXML::GetRequiredStringAttribute(p_pAffectorScaleElement, "function", strAffectorFunction);
+	const char* cAffectorFunction = strAffectorFunction.c_str();
+	if (strcmp(cAffectorFunction, "lifetime") == 0)
+	{
+		eAffectorFunction = Affector::AFFECTOR_FUNCTION::AF_LIFETIME;
+	}
+
+	float fScale = 0.0f;
+	EasyXML::GetRequiredFloatAttribute(p_pAffectorScaleElement, "amount", fScale);
+	m_vAffectors.push_back(new AffectorScale(eAffectorFunction, fScale));
+}
+
+void HeatStroke::Emitter::InitAffectorVelocity()
+{
+	m_vAffectors.push_back(new AffectorVelocity());
+}
+
+void HeatStroke::Emitter::InitParticles()
+{
+	Particle* pParticles = new Particle[m_iNumParticles];
+	for (int i = 0; i < m_iNumParticles; ++i)
+	{
+		AddParticleToPool(&pParticles[i]);
+	}
+}
+
+void HeatStroke::Emitter::AddParticleToPool(Particle* p_pParticle)
+{
+	p_pParticle->m_pPrev = nullptr;
+	p_pParticle->m_pNext = m_pFreeList;
+	m_pFreeList = p_pParticle;
```

Comment:
You're not setting m_pFreeList.prev = p_pParticle, so everyone's prev will point to nullptr